docs/en/plugins/README.md

```diff
@@ -0,0 +1,413 @@
+ ```
+**Associate that tag with a discussion:**
+```
+> dt = DiscussionTag.create(discussion: Discussion.last, tag: t)
+```
+**Ensure the discussion knows about its new tag:**
+```
+> Discussion.last.tags.first
+ => #<Tag id: 2, group_id: 33707, name: "Test tag", color: "#cccccc", created_at: "2016-03-07 01:40:08", updated_at: "2016-03-07 01:40:08">
+```
+
+Sweet!
+
+### PART 3: Create an API endpoint
+Now that we’ve got some models and a place to store them, the next step is to make that information available via our API. If you’re going to be writing plugins which expose info through the API, it’s strongly suggested you read up on the [snorlax gem](https://github.com/loomio/snorlax) that we’re using for our API controllers, and take a nice look through some of the existing controllers in `app/controllers/api`. It’s a bit of a DSL, but once you get the hang of it, writing new endpoints is a breeze!
+
+##### 1. Create a route
+First, let’s expose an endpoint, ‘/api/v1/discussion_tags.json’, which will route to the index action in a DiscussionTagsController:
+
+```
+# plugins/loomio_tags/plugin.rb (within the setup! method)
+plugin.use_route :get, '/discussion_tags', 'discussion_tags#index'
+```
+
+##### 2. Create a controller
+Now, we’ll define our index action in a new controller:
+
+```
+# plugins/loomio_tags/controllers/discussion_tags_controller.rb
+class ::API::DiscussionTagsController < API::RestfulController
+  def index
+    instantiate_collection { |collection| collection.where('discussions.key': params[:discussion_keys]) }
+    respond_with_collection
+  end
+
+  private
+
+  def accessible_records
+    resource_class.joins(:discussion)
+                             .where('discussions.id': Queries::VisibleDiscussions.new(user: current_user).pluck(:id))
+  end
+end
+```
+This may look a little confusing, but here’s the gist:
+- **accessible_records** defines the DiscussionTags the current user has access to, namely those associated with discussions which the user can see
+- **instantiate_collection** passes in those accessible records as a collection, and assigns the result of the block to @discussion_tags
+- **respond_with_collection** takes @discussion_tags and serializes it according to our API standards (more on that [here](https://github.com/loomio/tech-manual/blob/master/using_the_loomio_api.md#api-response-information))
+
+##### 3. Create a serializer
+Finally, we’ll define how those discussion tags we’re responding with appear in our API response. A simplified version:
+
+```
+# plugins/loomio_tags/serializers/discussion_tag_serializer.rb
+class DiscussionTagSerializer < ActiveModel::Serializer
+  embed :ids, include: true
+  attributes :id, :tag_name, :tag_color, :discussion_id
+end
+```
+
+You can check out some of the other serializers we've written in [app/serializers](https://github.com/loomio/loomio/tree/master/app/serializers)
```

Comment:
You can check out some of the other serializers we've written in [app/serializers](https://github.com/loomio/loomio/tree/master/app/serializers)**.** [add fullstop]